sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -190,7 +208,12 @@ public HttpHeaders getHeaders() {
                 if (!reactorNettyConnection.isDisposed()) {
                     reactorNettyConnection.channel().eventLoop().execute(reactorNettyConnection::dispose);
                 }
-            }).map(ByteBuf::nioBuffer);
+            }).map(byteBuf -> {
+                if (this.disableBufferCopy) {
+                    return byteBuf.nioBuffer();
+                }
+                return deepCopyBuffer(byteBuf);
```

Comment:
Reduce this to a one-liner: `return disableBufferCopy ? byteBuf.nioBuffer() : deepCopyBuffer(byteBuf);`